### ISSUE TITLE
Document that 'cargo package' always rebuilds all dependencies

### ISSUE THREAD

**Issue by Wilfred** [rust-lang/cargo#3062]
Every time I invoke `cargo package`, everything gets rebuilt -- even dependencies.

This is awkward when fixing missing entries in Cargo.toml:

```
$ cargo package
warning: manifest has no license or license-file. See http://doc.crates.io/manifest.html#package-metadata for more info.
   Packaging bfc v1.7.0 (file:///home/wilfred/projects/bfc)
   Verifying bfc v1.7.0 (file:///home/wilfred/projects/bfc)
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling libc v0.2.15
   Compiling regex-syntax v0.3.4
...
# OK, I've changed Cargo.toml. Let me try again to see if the warning has gone:
$ cargo package
$ cargo package
   Packaging bfc v1.7.0 (file:///home/wilfred/projects/bfc)
   Verifying bfc v1.7.0 (file:///home/wilfred/projects/bfc)
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling bitflags v0.5.0
   Compiling winapi v0.2.8
   Compiling semver v0.1.20
   Compiling matches v0.1.2 # looks like everything is being compiled again
```

Seems like the inverse of #2799.

**Comment by alexcrichton** [comment]
Ah yeah I believe after a successful package or when we're packaging we blow away entire directories to ensure that everything works from scratch. That's intentional currently to ensure that everything works from a pristine state, but maybe we could add a flag to not blow away existing directories?

**Comment by Wilfred** [comment]
Aha, OK. Does that make sense for dependencies too? (Is it possible to dirty your dependencies?)

This is sufficiently rare that I don't see much value in a flag. I think the behaviour is worth documenting though -- have I missed it, or should we add it?

**Comment by alexcrichton** [comment]
Perhaps no, but dependencies can often do weird things like leave around build artifacts which get stale, so it can be seen as good hygiene to always start from scratch.

Seems reasonable to document at least!

**Comment by phoepsilonix** [comment]
For example, if a Linux distributor is creating a package, they may wish to have the ability to check the list of licences for the relevant crates as a standard feature of cargo.
